packages/base/components/Embed/EmbedGateway.vue

```diff
@@ -0,0 +1,336 @@
+  }
+
+  // open the gate when there is no iframe/script embed, but other code
+  // rendered such as audio, video or plain HTML
+  opened.value = !iframe.value.src && !script.value.src;
+};
+
+const checkConsentAndOpenEmbed = () => {
+  const consents = acceptedServices.value;
+  const providerHasConsent = !!consents?.includes(provider.value?.name);
+
+  if (providerHasConsent) {
+    opened.value = true;
+  }
+};
+
+const consentAllEmbeddedContent = () => {
+  if (consentRequired.value) {
+    checkedServices.value = services.map((service) => service.name);
+  } else {
+    const allThirdPartyContentServices = services
+      .filter((s) => s.purposes.includes("thirdPartyContent"))
+      .flat()
+      .map((service) => service.name);
+    checkedServices.value = [...checkedServices.value].concat(
+      allThirdPartyContentServices,
+    );
+
+    saveConsents();
+  }
+};
+
+const consentThisProvider = () => {
+  checkedServices.value = [...checkedServices.value, provider.value.name];
+
+  if (!consentRequired.value) {
+    saveConsents();
+  }
+};
+
+const saveConsents = () => {
+  acceptOnly([...checkedServices.value]);
+  checkConsentAndOpenEmbed();
+};
+</script>
+
+<template>
+  <client-only>
+    <div class="h-100">
+      <slot v-if="opened" class="embed-gateway-opened" />
+      <div
+        v-else-if="provider"
+        class="container notification-overlay"
+        :class="{ 'mw-100': embedCode }"
+      >
+        <div class="row position-relative">
+          <div class="col thumbnail-background mx-auto h-100 position-absolute">
+            <!-- Add MediaCard when used for items -->
+            <div
+              class="icon-multimedia h-100 d-flex align-items-center justify-content-center"
+            />
+          </div>
+          <div
+            class="col notification-content mx-auto position-relative"
+            :style="{
+              'min-height': !!iframe.height && iframe.height,
+              width: !!iframe.width && iframe.width,
+            }"
+          >
+            <p class="message">
+              {{ $t("embedNotification.message", { provider: providerName }) }}
+            </p>
+            <button
+              class="accept-all-button btn btn-light mb-2"
+              :data-bs-toggle="consentRequired && 'modal'"
+              :data-bs-target="`#${cookieModalId}`"
+              @click="consentAllEmbeddedContent"
+            >
+              {{ $t("embedNotification.loadAllEmbeddedContent") }}
+            </button>
+            <i18n-t
+              keypath="embedNotification.ofThirdPartyServices"
+              tag="p"
+              scope="global"
+            >
+              <button
+                class="btn btn-link"
+                data-bs-toggle="modal"
+                :data-bs-target="`#${cookieModalId}`"
+              >
+                {{ $t("embedNotification.viewFullList") }}
+              </button>
+            </i18n-t>
+            <!-- TODO: lazy load CookiesModal -->
```

Comment:
Do this already as otherwise there may be multiple such components loaded, all with the same ID, despite potentially never being shown.